diff --git a/esm/stateFromElement.js b/esm/stateFromElement.js
index eccabfd5230490173bd92c25d73c55a1fd9529fc..ffeb566cf1699a5211009f44402728351c025d53 100644
--- a/esm/stateFromElement.js
+++ b/esm/stateFromElement.js
@@ -50,6 +50,7 @@ var ELEM_ATTR_MAP = {
     height: 'height'
   }
 };
+export const STYLE_SEPARATOR = " ";
 
 var getEntityData = function getEntityData(tagName, element) {
   var data = {};
@@ -373,7 +374,7 @@ function () {
         switch (customInline.type) {
           case 'STYLE':
             {
-              style = style.add(customInline.style);
+              customInline.style?.split(STYLE_SEPARATOR).forEach(customStyle => style = style.add(customStyle));
               break;
             }
 
diff --git a/lib/stateFromElement.js b/lib/stateFromElement.js
index e814c0500a48054f4a891bb28e4fd899718e102b..946762f3f61256ee7e5f505a35334777e5ec1e82 100644
--- a/lib/stateFromElement.js
+++ b/lib/stateFromElement.js
@@ -66,6 +66,7 @@ var ELEM_ATTR_MAP = {
     height: 'height'
   }
 };
+const STYLE_SEPARATOR = " ";
 
 var getEntityData = function getEntityData(tagName, element) {
   var data = {};
@@ -389,7 +390,7 @@ function () {
         switch (customInline.type) {
           case 'STYLE':
             {
-              style = style.add(customInline.style);
+              customInline.style?.split(STYLE_SEPARATOR).forEach(customStyle => style = style.add(customStyle));
               break;
             }
 
@@ -670,3 +671,4 @@ function stateFromElement(element, options) {
 
 var _default = stateFromElement;
 exports["default"] = _default;
+exports.STYLE_SEPARATOR = STYLE_SEPARATOR;
